composer/callbacks/speed_monitor.py

```diff
@@ -360,10 +361,18 @@ def batch_end(self, state: State, logger: Logger):
         # Log the time
         # `state.timestamp` excludes any time spent in evaluation
         train_wct = state.timestamp.total_wct.total_seconds()
+        secs_per_step = 0
+        if len(self.history_wct) > 1:
+            secs_per_step = self.history_wct[-1] - self.history_wct[-2]
```

Comment:
This should probably average over history_wct as all metrics respect `window_size`. Something like:
`elapsed_wct = (self.history_wct[-1] - self.history_wct[0]) / (len(history-wct)-1)`